vendor/k8s.io/kubernetes/pkg/util/iptables/iptables.go

```diff
@@ -335,6 +359,19 @@ func (runner *runner) restoreInternal(args []string, data []byte, flush FlushFla
 		args = append(args, "--counters")
 	}
 
+	// Grab the iptables lock to prevent iptables-restore and iptables
+	// from stepping on each other.  iptables-restore 1.6.2 will have
+	// a --wait option like iptables itself, but that's not widely deployed.
+	fd, err := grabIptablesLock()
+	if err != nil {
+		return err
+	}
+	defer func(lockFile *os.File) {
+		if lockFile != nil {
+			fd.Close()
+		}
+	}(fd)
```

Comment:
@danwinship it can be nil because:

```
	// First attempt to acquire the iptables 1.6.x lockfile
	fd, err := os.OpenFile("/run/xtables.lock", os.O_CREATE, 0600)
	if err != nil {
		// Ignore errors opening the lockfile like xtables_lock() does
		return nil, nil
	}
```

we don't have to duplicate iptables behavior, but if we do, it can be nil and still succeed.